test/integration/host_js_test.rb

```diff
@@ -0,0 +1,472 @@
+      click_button('Submit')
+    end
+  end
+
+  describe "NIC modal window" do
+    describe "editing interfaces" do
+      test "click on edit opens modal" do
+        go_to_interfaces_tab
+
+        # edit the primary interface
+        table.first(:button, 'Edit').click
+
+        assert modal.find('.modal-content').visible?, "Modal window was shown"
+        assert modal.find('.interface_primary').checked?, "Primary checkbox is checked"
+        assert modal.find('.interface_provision').checked?, "Provision checkbox is checked"
+
+        modal.find(:button, "Cancel").click
+
+        # test column content
+        assert table.find('td.identifier').has_content?('')
+        assert table.find('td.type').has_content?('Interface')
+        assert table.find('td.mac').has_content?('')
+        assert table.find('td.ip').has_content?('')
+        assert table.find('td.fqdn').has_content?('')
+      end
+
+      test "ok button updates overview table" do
+        go_to_interfaces_tab
+
+        # edit the primary interface
+        table.first(:button, 'Edit').click
+
+        modal.find('.interface_identifier').set('eth0')
+        modal.find('.interface_mac').set('11:22:33:44:55:66')
+        modal.find('.interface_ip').set('10.32.8.3')
+        modal.find('.interface_name').set('name')
+
+        modal.find(:button, "Ok").click
+
+        assert table.find('td.identifier').has_content?('eth0')
+        assert table.find('td.type').has_content?('Interface')
+        assert table.find('td.mac').has_content?('11:22:33:44:55:66')
+        assert table.find('td.ip').has_content?('10.32.8.3')
+        assert table.find('td.fqdn').has_content?('')
+      end
+    end
+
+    describe "adding interfaces" do
+      test "click on add opens modal" do
+        go_to_interfaces_tab
+
+        assert_interface_change(0) do
+          page.find(:button, '+ Add Interface').click
+
+          assert modal.find('.modal-content').visible?, "Modal window was shown"
+          refute modal.find('.interface_primary').checked?, "Primary checkbox is unchecked by default"
+          refute modal.find('.interface_provision').checked?, "Provision checkbox is unchecked by default"
+
+          modal.find(:button, "Cancel").click
+        end
+      end
+
+      test "ok button adds new interface" do
+        go_to_interfaces_tab
+
+        assert_interface_change(1) do
+          page.find(:button, '+ Add Interface').click
+          modal.find(:button, "Ok").click
+        end
+      end
+
+      test "setting primary updates host name" do
+        go_to_interfaces_tab
+
+        # edit the primary interface
+        table.first(:button, 'Edit').click
+
+        domain = domains(:mydomain)
+
+        modal.find('.interface_name').set('name')
+        modal.find('.interface_domain').select(domain.name)
+
+        modal.has_select?('.interface_subnet', :options => domain.subnets.map(&:to_label))
+        modal.has_select?('.interface_domain', :selected => domain.name)
```

Comment:
These two has_select? lines appear to be taking a very long time to execute, about 30 seconds each, causing:

    09:39:26 HostJSTest::NIC modal window::adding interfaces#test_0003_setting primary updates host name = 65.35 s = .
    09:40:31 HostJSTest::NIC modal window::adding interfaces#test_0004_selecting domain updates subnet list = 64.95 s = .

I'm even seeing it locally, resulting in this stack trace:

<pre>
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/poltergeist-1.9.0/lib/capybara/poltergeist/web_socket_server.rb:80:in `call'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/poltergeist-1.9.0/lib/capybara/poltergeist/web_socket_server.rb:80:in `recv'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/poltergeist-1.9.0/lib/capybara/poltergeist/web_socket_server.rb:80:in `receive'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/poltergeist-1.9.0/lib/capybara/poltergeist/web_socket_server.rb:97:in `send'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/poltergeist-1.9.0/lib/capybara/poltergeist/server.rb:38:in `send'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/poltergeist-1.9.0/lib/capybara/poltergeist/browser.rb:344:in `command'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/poltergeist-1.9.0/lib/capybara/poltergeist/browser.rb:67:in `find_within'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/poltergeist-1.9.0/lib/capybara/poltergeist/node.rb:17:in `command'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/poltergeist-1.9.0/lib/capybara/poltergeist/node.rb:34:in `find'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/poltergeist-1.9.0/lib/capybara/poltergeist/node.rb:38:in `find_xpath'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/capybara-2.6.2/lib/capybara/node/base.rb:107:in `find_xpath'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/capybara-2.6.2/lib/capybara/query.rb:110:in `block in resolve_for'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/capybara-2.6.2/lib/capybara/node/base.rb:80:in `synchronize'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/capybara-2.6.2/lib/capybara/query.rb:106:in `resolve_for'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/capybara-2.6.2/lib/capybara/node/matchers.rb:94:in `block in assert_selector'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/capybara-2.6.2/lib/capybara/node/base.rb:84:in `synchronize'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/capybara-2.6.2/lib/capybara/node/matchers.rb:93:in `assert_selector'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/capybara-2.6.2/lib/capybara/node/matchers.rb:39:in `has_selector?'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/capybara-2.6.2/lib/capybara/node/matchers.rb:387:in `has_select?'
test/integration/host_js_test.rb:400:in `block (3 levels) in <class:HostJSTest>'
</pre>